src/functions-and-arrays.js

```diff
@@ -1,41 +1,141 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(n1,n2) {
+  if (n1 > n2){
+    return n1;
+  } else {
+    return n2;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  let longestWord;
+  if (words.length === 0){
+    return null;
+
+  } else {
+    for (let i=0; i < words.length; i++){
+      let word = words[i];
+      if (!longestWord || longestWord.length < word.length){
+      longestWord = word;
+  }
+    } 
+    return longestWord;
+  }
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0
+  if (numbers.length === 0){
+    return 0;
+  } else {
+    for (let i = 0; i<numbers.length; i++){
+      let number = numbers[i]
+      sum += number;
+      }
+  }
+  return sum
 
+  }
+  
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(mixedArr) {
+  let countChar = 0;
+  let countNum = 0;
+  let countBol = 0;
+  
+  if (mixedArr.length === 0) {
+    return 0;
+  } else {
+    for (let i = 0; i < mixedArr.length; i++) {
+      let element = mixedArr[i];
+      if (typeof element === 'number') {
+        countNum += element;
+      } else if (typeof element === 'string') {
+        countChar += element.length;
+      } else if (typeof element === 'boolean'){
+        countBol += element;
+      } else {
+        throw new Error("Unsupported data type sir or ma'am");
+      }
+    }
+  }
+  let sumNumChar = countChar + countNum+ countBol;
+  return sumNumChar;
```

Comment:
```suggestion
  let count = 0;
  if (mixedArr.length === 0) {
    return 0;
  } else {
    for (let i = 0; i < mixedArr.length; i++) {
      let element = mixedArr[i];
      if (typeof element === 'number' || typeof element === 'boolean') {
        count += element;
      } else if (typeof element === 'string') {
        count += element.length;
      } else {
        throw new Error("Unsupported data type sir or ma'am");
      }
    }
  }
  return count;
```

Bien!, entiendo perfectamente lo de las 3 variables, pero fijate la propia definicion del metodo te dice que es una suma, es decir que todo tendra que ser un numero!, por ello, si te creas una unica variable que sea de contador lo tendrias, ahora bien, veras que he puesto los numeros y los booleanos juntos, esto es porque si tu intentas sumar un booleano hace lo siguiente, true = 1 y false = 0, por ello si intentas hacer const x = 1 + true, x valdria 2! :D